outline/connectivity/connectivity.go

```diff
@@ -48,7 +48,7 @@ type reachabilityError struct {
 // the current network. Parallelizes the execution of TCP and UDP checks, selects the appropriate
 // error code to return accounting for transient network failures.
 // Returns an error if an unexpected error ocurrs.
-func CheckConnectivity(client *outline.Client) (int, error) {
+func CheckConnectivity(client *outline.Client) (neterrors.Error, error) {
```

Comment:
I think we can just return `error`, and use [`errors.Is`](https://pkg.go.dev/errors#Is) to determine whether it is our `neterrors.Error` object.

Reply:
I agree!
Using error unwrapping would be much cleaner, but that would break the existing API, so I left it as is for now.